gm4_double_doors/generate.py

```diff
@@ -0,0 +1,99 @@
+from beet import Context, Function, Advancement, Structure
+from typing import Dict, List
+from dataclasses import dataclass
+
+NAMESPACE = 'gm4_double_doors'
+TEMPLATE_FOLDER = 'template'
+TEMPLATE_PLACEHOLDER = '$material_name$'
+
+
+@dataclass
+class Clone:
+    name: str
+
+    advancements: Dict[str, Advancement]
+    functions: Dict[str, Function]
+    structures: Dict[str, Structure]
+
+
+def beet_default(ctx: Context):
+
+    # list of supported wood types. Add new wood types here.
+    material_names: List[str] = ['acacia', 'bamboo', 'birch', 'cherry', 'crimson', 'dark_oak', 'jungle', 'mangrove', 'oak', 'spruce', 'warped']
```

Comment:
We can automatically pull this from the `beet.contrib.vanilla` Perhaps taking the wood types from `#minecraft:wooden_doors`? There might be a better source of just the wood types without extra *stuff* like the `_door`